src/runtime/startup/smaster.go

```diff
@@ -106,23 +114,33 @@ func SMaster(socket C.int, instanceSocket C.int, config *C.struct_cConfig, jsonC
 }
 
 func main() {
+	loglevel := os.Getenv("SINGULARITY_MESSAGELEVEL")
+
+	os.Clearenv()
```

Comment:
Is necessary to have here too since it's added in C code, or it's just in case?

Reply:
Yes I was surprise too, Go code don't use the same environment variable stack as C, seems to create/copy its own :/

Reply:
Go is a funny guy